src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -1030,6 +1056,26 @@ public void removeDialog(String dialogId) {
         }
         dialogTable.remove(dialogId);
     }
+    
+    /**
+     * Adds a new dialog injection listener to this transaction.
+     * 
+     * @param newListener
+     *          Listener to add.
+     */
+    public void addDialogInjectionListener(SIPDialogInjectionListener newListener) {
+        dialogInjectionListeners.add(newListener);
```

Comment:
this is a write scenario into a nonSync collection, do you expect concurrent access?if yes, how do you plan to prevent issues?